### ISSUE TITLE
Investigate video playback on iPad and Kindle Fire

### ISSUE THREAD

**Issue by jmcglone** [mlibrary/heliotrope#401]
Current behavior: When I view a video asset page on an iPad, Kindle Fire, iPhone, or Safari, the video asset does not playback.

Expected behavior: When I view a video asset page on an iPad or Kindle Fire, the video asset should have playback controls and play when I touch the play button.

- [ ] Video plays back on iPad, Kindle, and iOS (when content-type is set correctly)
- [ ] Tested across other touch screen and webkit devices in browserstack and works (when content-type is set correctly)

**Comment by sethaj** [comment]
Could be related to this? It looks like we have multiple sources in our markup.

http://stackoverflow.com/a/7207710

**Comment by jmcglone** [comment]
This issue also extends to iPhones and Safari

**Comment by jmcglone** [comment]
So far I have attempted the following to resolve the issue:

1. [Only use a single `source` (mp4) in the `video` element](http://stackoverflow.com/a/5077687).
2. [Placing the `src` directly in the `video` element](http://stackoverflow.com/a/5077687). 
3. [Changing the `controls` attribute to `true`](http://stackoverflow.com/a/25670764).
4. [Ensure my apache server configuration is using the correct mime-types](http://stackoverflow.com/a/5080688).

Still need to try:

- [MOOV atom should be at the beginning of the mp4](http://stackoverflow.com/a/28512437). [See also](http://stackoverflow.com/a/38508727)
- [Check to ensure apache supports byte-range requests for video](http://stackoverflow.com/a/28779883)

**Comment by jmcglone** [comment]
After inspecting the Headers, it looks like mp4 files content-type is incorrectly set to 'application/mp4'. They should be set to 'video/mp4'.

```
m-c02mjbdzf8j2:~ jmcglone$ curl -I http://localhost:3000/downloads/mw22v555t?file=mp4
HTTP/1.1 200 OK
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Accept-Ranges: bytes
Content-Disposition: inline; filename="t-mp4.mp4"
Content-Transfer-Encoding: binary
Content-Type: application/mp4
Cache-Control: private
X-Request-Id: aab1db4f-d682-4076-b589-e1d9ce697bbf
X-Runtime: 0.018895
Content-Length: 35982841
```

Also, content-type is not correct for webm files - it is set to 'audio/webm' and should be 'video/webm':

```
m-c02mjbdzf8j2:~ jmcglone$ curl -I http://localhost:3000/downloads/mw22v555t?file=webm
HTTP/1.1 200 OK
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Accept-Ranges: bytes
Content-Disposition: inline; filename="t-webm.webm"
Content-Transfer-Encoding: binary
Content-Type: audio/webm
Cache-Control: private
X-Request-Id: 9165f21c-ba3b-49b4-910e-c6513b388a2e
X-Runtime: 0.094042
Content-Length: 31571458
```

**Comment by jmcglone** [comment]
Could it be related to this bug in CC?

https://github.com/projecthydra/curation_concerns/issues/258

**Comment by jmcglone** [comment]
The content-type sent to the browser is incorrectly set by the download_behavior controller. A fix is proposed int #427.

When the content-type is set correctly, mp4 files do playback in Safari, iOS, and Kindle. I confirmed this by pointing the `video` element's `source` to an external mp4 file hosted on an external server that was correctly setting the content-type for the file to `video/mp4`.

**Comment by jmcglone** [comment]
I believe the problem is related to rails being unable to do byte range requests. See http://stackoverflow.com/questions/33744633/safari-not-loading-html5-video-in-rails-app and http://stackoverflow.com/questions/6759426/rails-media-file-stream-accept-byte-range-request-through-send-data-or-send-file and http://stackoverflow.com/questions/28008564/streaming-mp4-in-chrome-with-rails-nginx-and-send-file?noredirect=1&lq=1
